Remotion/Data/DomainObjects/Infrastructure/TypePipe/ClassDefinitionProvider.cs

```diff
@@ -22,22 +22,22 @@
 namespace Remotion.Data.DomainObjects.Infrastructure.TypePipe
 {
   /// <summary>
-  /// Implements <see cref="ITypeDefinitionProvider"/> by retrieving the <see cref="ClassDefinition"/> of the domain object type from the current 
+  /// Implements <see cref="IClassDefinitionProvider"/> by retrieving the <see cref="ClassDefinition"/> of the domain object type from the current 
   /// mapping configuration.
   /// </summary>
   /// <threadsafety static="true" instance="true"/>
-  [ImplementationFor(typeof(ITypeDefinitionProvider), Lifetime = LifetimeKind.Singleton, RegistrationType = RegistrationType.Single)]
-  public class TypeDefinitionProvider : ITypeDefinitionProvider
+  [ImplementationFor(typeof(IClassDefinitionProvider), Lifetime = LifetimeKind.Singleton, RegistrationType = RegistrationType.Single)]
+  public class ClassDefinitionProvider : IClassDefinitionProvider
   {
-    public ClassDefinition? GetTypeDefinition (Type domainObjectType)
+    public ClassDefinition? GetClassDefinition (Type domainObjectType)
     {
       ArgumentUtility.CheckNotNull("domainObjectType", domainObjectType);
 
       var mappingConfiguration = MappingConfiguration.Current;
       if (!mappingConfiguration.ContainsTypeDefinition(domainObjectType))
```

Comment:
It also return the class definition so I would keep the current name

Reply:
No, I was referring to using the ClassDefinition-only implementation `ContainsClassDefinition` instead of the any-TypeDefinition `ContainsTypeDefinition` method. By testing for any TypeDefinition instead of just ClassDefinition, the implementation is at least strange if not buggy. Don't see the entire method in the current github quote so I don't know the full damage right now.